lightning/src/util/config.rs

```diff
@@ -195,18 +208,8 @@ pub struct ChannelConfig {
 	///
 	/// Default value: false.
 	pub announced_channel: bool,
-	/// When set, we commit to an upfront shutdown_pubkey at channel open. If our counterparty
-	/// supports it, they will then enforce the mutual-close output to us matches what we provided
-	/// at intialization, preventing us from closing to an alternate pubkey.
-	///
-	/// This is set to true by default to provide a slight increase in security, though ultimately
-	/// any attacker who is able to take control of a channel can just as easily send the funds via
-	/// lightning payments, so we never require that our counterparties support this option.
-	///
-	/// This cannot be changed after a channel has been initialized.
-	///
-	/// Default value: true.
-	pub commit_upfront_shutdown_pubkey: bool,
+	/// This value is moved to ChannelHandshakeConfig, optional here for old serialiization
+	pub commit_upfront_shutdown_pubkey: Option<bool>,
```

Comment:
Lets make this `pub(crate)` and default to `None`? That way its really only used internally and we'll start defaulting to using the new one instead by seeing its `None`.